cmd/ck-client/ck-client.go

```diff
@@ -47,6 +47,18 @@ func makeSession(sta *client.State, isAdmin bool) *mux.Session {
 				time.Sleep(time.Second * 3)
 				goto makeconn
 			}
+			err = remoteConn.(*net.TCPConn).SetKeepAlive(true)
+			if err != nil {
+				log.Errorf("Setting KeepAlive: %v\n", err)
+				time.Sleep(time.Second * 3)
+				goto makeconn
+			}
+			err = remoteConn.(*net.TCPConn).SetKeepAlivePeriod(time.Duration(15 * time.Second))
```

Comment:
In https://golang.org/pkg/net/#Dialer it says when Diailer.KeepAlive (defined on line 35) is zero, it defaults to 15 seconds. Is this explicit setting still necessary?

Reply:
I like to be explicit about the timing in case of compiling with different go versions.

Reply:
I think this behaviour was added in Go 1.12: https://github.com/golang/go/issues/23459. That's the minimum version needed to compile Cloak. In order to be explicit it may be better to set the KeepAlive field on line 35. Errors will be caught on line 43